lib/hacks/grid-utils.js

```diff
@@ -213,22 +221,34 @@ function parseTemplate({
 
             // area
             if (type === 'string') {
+                result = verifyRowSize(result);
                 result.areas.push(value);
+                result.rowProps = true;
+                result.sizedRow = false;
             }
 
             // values and function
             if (type === 'word' || type === 'function') {
                 result[result.key].push(parser.stringify(node));
+                if (result.rowProps &&
+                    !(value.startsWith('[') && value.endsWith(']'))) {
+                    result.sizedRow = true;
+                }
             }
 
             // devider(/)
             if (type === 'div' && value === '/') {
                 result.key = 'columns';
+                result = verifyRowSize(result);
+                result.rowProps = false;
             }
 
```

Comment:
Maybe it is enough to compare the length of areas and rows. And if the length of rows is less then the length of areas it needed to push `auto` to rows array.